test/test_deployment.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+# Replication Controller entity tests
+class TestDeployment < MiniTest::Test
+  def test_get_from_json_v1
+    stub_request(:get, %r{/deployment})
+      .to_return(body: open_test_file('deployment.json'),
+                 status: 200)
+
+    Kubeclient::Deployment.new
+    client = Kubeclient::Client.new 'http://localhost:8080/apis', 'extensions/v1beta1'
```

Comment:
> @henders at first glance I'd say that we would like to formalize better how to reach extensions and their api versioning (according to the new api groups in kubernetes). This line works but has the feeling of a temporary workaround.

I agree, we need a better solution to work with multiple versions and experimental features.